src/frequenz/sdk/core/timeseries.py

```diff
@@ -0,0 +1,71 @@
+"""
+Common model definitions used throughout the SDK.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+from __future__ import annotations
+
+import datetime
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class TimeSeriesId:
+    """ID for a single metric of a specific component, e.g. 123_active_power."""
+
+    component_id: int
+    metric_id: str
+
+    def __str__(self) -> str:
+        """Return string representation of a time series id.
+
+        Returns:
+            string representation of a time series id
+        """
+        return f"{self.component_id}_{self.metric_id}"
+
+    @staticmethod
+    def from_str(name: str) -> TimeSeriesId:
+        """Instantiate a time series id object from a string.
+
+        Args:
+            name: string representation of a time series id
+
+        Returns:
+            time series id
+
+        Raises:
+            ValueError: if provided string has an invalid format
+            TypeError: if the first segment of the string is not an integer
+        """
+        parts = name.split("_")
+        if len(parts) < 2:
+            raise ValueError(
+                f"String representation of a time series id must have a "
+                f"least two segments separated by underscore, but {parts}"
+                f" was provided"
+            )
+
+        try:
+            component_id = int(parts[0])
+        except TypeError as err:
+            raise TypeError(
+                f"First segment of the string representation of a time "
+                f"series id must be an integer, but {parts[0]} was "
+                f"provided"
+            ) from err
+
+        metric_id = "_".join(parts[1:])
+        return TimeSeriesId(component_id=component_id, metric_id=metric_id)
+
+
+@dataclass(frozen=True)
+class Sample:
+    """Describes a single observed floating point value at a specific timestamp."""
+
+    timestamp: datetime.datetime
+    value: float
```

Comment:
I was wondering where it should go.. first I put all these dataclasses into `/core/common.py`, but @leandro-lucarella-frequenz suggested that the `TimeSeriesId` should go to `timeseries.py` and I moved `Sample` here too. Maybe `/core/models.py` or `/core/common.py`?

I guess it could also go into this directory `/data_ingestion/resampling/`, e.g. `/data_ingestion/resampling/common.py`.

Do you have any suggestions?